packages/vats/test/bootstrapTests/supports.js

```diff
@@ -488,4 +498,96 @@ export const makeSwingsetTestKit = async (
     timer,
   };
 };
+
 /** @typedef {Awaited<ReturnType<typeof makeSwingsetTestKit>>} SwingsetTestKit */
+
+export const makeTestContext = async t => {
+  console.time('DefaultTestContext');
+  /** @type {SwingsetTestKit} */
+  const swingsetTestKit = await makeSwingsetTestKit(t, 'bundles/vaults', {
+    configSpecifier: PLATFORM_CONFIG,
+  });
+
+  const { runUtils, storage } = swingsetTestKit;
+  console.timeLog('DefaultTestContext', 'swingsetTestKit');
+  const { EV } = runUtils;
+
+  // Wait for ATOM to make it into agoricNames
+  await EV.vat('bootstrap').consumeItem('vaultFactoryKit');
+  console.timeLog('DefaultTestContext', 'vaultFactoryKit');
+
+  await eventLoopIteration();
+
+  // has to be late enough for agoricNames data to have been published
+  const agoricNamesRemotes = makeAgoricNamesRemotesFromFakeStorage(
+    swingsetTestKit.storage,
+  );
+  agoricNamesRemotes.brand.ATOM || Fail`ATOM brand not yet defined`;
+  console.timeLog('DefaultTestContext', 'agoricNamesRemotes');
+
+  const walletFactoryDriver = await makeWalletFactoryDriver(
+    runUtils,
+    storage,
+    agoricNamesRemotes,
+  );
+  console.timeLog('DefaultTestContext', 'walletFactoryDriver');
+
+  console.timeEnd('DefaultTestContext');
+
+  const buildProposal = makeProposalExtractor({
+    childProcess: processAmbient,
+    fs: fsAmbient.promises,
```

Comment:
> modules should not export powerful objects (for example, objects that close over fs or process.env)
 -- [OCap discipline](https://github.com/Agoric/agoric-sdk/wiki/OCap-Discipline)

Please let's pass powers such as `fs` and `childProcess` into `makeTestContext` explicitly, if we're going to export it.

(yes, it's "just" test code, but I don't think this package as an `exports` thing that prevents unwanted deep imports, so this is actually part of the public API).

Reply:
I converted the PR to DRAFT to make it clear that it's not intended to be merged. The point of the PR is to show that an upgrade from current state to post-ZCF would work. It's a rough copy of #7966, which is intended to merge, but has to start from the post-ZCF world.

It doesn't seem worth spending effort to clean this up, as long as it's convincing. Let me know if you'd like me to do it anyway.

Reply:
ah. thanks for clarifying. sorry to bother you.